store/index.js

```diff
@@ -116,12 +133,64 @@ export const actions = {
 		p_context.commit("addColumnCategorization", {
 
 			bColor: categorizationInfo.current.bColor,
-			dataDictionaryColumn: p_columnName,
+			dataDictionaryColumn: p_clickData.column,
 			fColor: categorizationInfo.current.fColor,
+			primaryKey: p_clickData["primary-key"],
 			tsvCategory: categorizationInfo.current.category
 		});
 	},
 
+	retrievePalette(p_context) {
+
+		// 0. New color lists
+		let backgroundColors = [];
+		let foregroundColors = [];
+
+		// 1. Wipe the palette
+		p_context.commit("clearPalette");
+
+		// 2. Go through global stylesheets until we find the paintClasses
+		for ( let sheetID in document.styleSheets ) {
+
+			for ( let ruleID in document.styleSheets[sheetID].cssRules ) {
+
+				let paintClasses = p_context.state.painting.classes;
+				for ( let paintClass in paintClasses ) {
+
+					// A. Make sure the CSS ruleset is an object containing a CSS string with the desired paint class
+					if ( typeof (document.styleSheets[sheetID].cssRules[ruleID]) === "object" &&
+						 "cssText" in document.styleSheets[sheetID].cssRules[ruleID] &&
+						 -1 != document.styleSheets[sheetID].cssRules[ruleID].cssText.indexOf(paintClasses[paintClass]) ) {
+
+						// I. Parse the CSS string for this class into an object
+						let cssProperties = parseCssString(
+							document.styleSheets[sheetID].cssRules[ruleID].cssText);
+
+						// II. Retrieve the background-color and color from the css object
+						let backgroundColor = getCssValue(
+							cssProperties,
+							"background-color"
+						);
+						let foregroundColor = getCssValue(
+							cssProperties,
+							"color"
+						);
+
+						// III. Save the background and foreground colors
+						backgroundColors.push(backgroundColor);
+						foregroundColors.push(foregroundColor);
+					}
+				}
+			}
+		}
+
+		// 3. Set the new palette
+		p_context.commit("setPalette", {
+			backgroundColors: backgroundColors,
+			foregroundColors: foregroundColors
+		});
+	},	
+
```

Comment:
If I understand this correctly, the entire above added section (`retrievePalette`) exists in order to regex parse the main.css stylesheet in order to retrieve the colors we have defined there and store them in a global store `backgroundColors` and `foregroundColors`. And if I also understand this correctly, the only reason we need to do that is because we keep track of what category a column has been assigned to by comparing it's current DOM style color attribute to the values in this list.

As I mentioned above, I really think we need to completely split colors and categorization and focus on keeping track of the categorization.